### PR TITLE
Created additional utilities for login

### PR DESCRIPTION
**Pull request by vglozinic**
Utilities for reset of forgotten password and checking if username exists in database.

### CODE REVIEW

File: src/main/java/model/UserModel.java [hunk 1]
```diff
@@ -28,17 +28,30 @@ public int checkParameters(Map<String, String[]> parameters) {
...
+			e.printStackTrace();
+		}
+		return result;
+	}
+	
+	public boolean checkUser(String username) {
+		boolean result = false;
+		ResultSet data = daoFactory.getUserDao().checkUser(username);
 		
-		if(data != null) {
```
Comment (gabilic) [inline src/main/java/model/UserModel.java]: Why is this line removed? I think `data` should be tested for `null` value to avoid an exception.

File: src/main/java/model/UserModel.java [hunk 2]
```diff
@@ -28,17 +28,30 @@ public int checkParameters(Map<String, String[]> parameters) {
...
-			try {
-				data.next();
-				if(Password.checkPassword(password, data.getString(2), data.getString(3))) {
-					result = data.getInt(1);
-				}
-			} catch (SQLException e) {
-				e.printStackTrace();
+		try {
```
Comment (gabilic) [inline src/main/java/model/UserModel.java]: I think `data` should be tested for `null` value before the `try-catch` block to avoid an exception.

File: src/main/java/model/UserModel.java (hunk 1 above)
Comment (ZMehanovic) [reply src/main/java/model/UserModel.java]: actually in java executeQuery method will never return a null so it is unnecessary. It should be checked with data.next()

File: src/main/java/model/UserModel.java (hunk 2 above)
Comment (ZMehanovic) [reply src/main/java/model/UserModel.java]: same as in the above example, in java executeQuery method will never return a null so it is unnecessary. It should be checked with data.next()

File: src/main/java/model/UserModel.java (hunk 1 above)
Comment (gabilic) [reply src/main/java/model/UserModel.java]: Not in case of the `SQLException` 😄

File: src/main/java/model/UserModel.java (hunk 2 above)
Comment (gabilic) [reply src/main/java/model/UserModel.java]: Not in case of the `SQLException` 😄
